src/ring_theory/roots_of_unity.lean

```diff
@@ -715,4 +716,79 @@ lemma nth_roots_one_eq_bind_primitive_roots {ζ : R} {n : ℕ} (hpos : 0 < n)
 
 end integral_domain
 
+section minimal_polynomial
+
+open minimal_polynomial
+
+variables {n : ℕ} {K : Type*} [field K] {μ : K} (h : is_primitive_root μ n) (hpos : 0 < n)
+
+include n μ h hpos
+
+/--`μ` is integral over `ℤ`. -/
+lemma is_integral : is_integral ℤ μ :=
+begin
+  use (X ^ n - 1),
+  split,
+  { exact (monic_X_pow_sub_C 1 (ne_of_lt hpos).symm) },
+  simp only [((is_primitive_root.iff_def μ n).mp h).left, eval₂_one, eval₂_X_pow, eval₂_sub,
+  sub_self],
+end
+
+/--The minimal polynomial of `μ` is monic. -/
+lemma is_monic_minimal_polynomial : (minimal_polynomial (is_integral h hpos)).monic :=
+(minimal_polynomial.monic (is_integral h hpos))
+
+/--The minimal polynomial of `μ` is primitive. -/
+lemma is_primitive_minimal_polynomial : is_primitive
+(minimal_polynomial (is_integral h hpos)) := monic.is_primitive
+(is_monic_minimal_polynomial h hpos)
+
+variables [char_zero K]
+
+/--`μ ` is integral over `ℚ`. -/
+lemma is_algebraic : _root_.is_integral ℚ μ :=
+@is_integral_of_is_scalar_tower ℤ ℚ K _ _ _ (algebra_int ℚ) _ _ add_comm_group.int_is_scalar_tower
+  μ (is_integral h hpos)
+
+/--The minimal polynomial of `μ` over `ℚ` is the same as the minimal polynomial over `ℤ`. -/
+lemma min_poly_rational_eq_min_poly_integer : minimal_polynomial (is_algebraic h hpos)
+   = map (int.cast_ring_hom ℚ) (minimal_polynomial (is_integral h hpos)) :=
+begin
+  refine (unique' (is_algebraic h hpos) _ _ _).symm,
+  { exact (is_primitive.int.irreducible_iff_irreducible_map_cast (is_primitive_minimal_polynomial h
+  hpos)).1 (irreducible (is_integral h hpos)) },
+  { have htower := is_scalar_tower.aeval_apply ℤ ℚ K μ (minimal_polynomial
+    (is_integral h hpos)),
+    rw [minimal_polynomial.aeval] at htower,
+    exact htower.symm },
+  { exact monic_map _ (is_monic_minimal_polynomial h hpos) }
+end
+
+/--If `P : polynomial ℤ` is primitive and has `μ` as root, then the minimal polynomial of `μ`
+over `ℤ` divides `P`. -/
+lemma minimal_polynomial_dvd {P : polynomial ℤ} (hprim : is_primitive P)
+  (hroot : aeval μ P = 0) : minimal_polynomial (is_integral h hpos) ∣ P :=
+begin
+  apply (is_primitive.int.dvd_iff_map_cast_dvd_map_cast (minimal_polynomial
+  (is_integral h hpos)) P _ hprim).2,
+  { rw [←min_poly_rational_eq_min_poly_integer h hpos],
+    apply minimal_polynomial.dvd,
+    have htower := is_scalar_tower.aeval_apply ℤ ℚ K μ P,
+    rw [hroot] at htower,
+    exact htower.symm },
+  { exact is_primitive_minimal_polynomial h hpos }
+end
+
+/--The minimal polynomial of `μ` divides `X ^ n - 1`. -/
+lemma minimal_polynomial_dvd_X_pow_sub_one :
+  minimal_polynomial (is_integral h hpos) ∣ X ^ n - 1 :=
+begin
+  apply minimal_polynomial_dvd h hpos (polynomial.monic.is_primitive
```

Comment:
I think you can golf this a little bit by directly applying the lemma that minimal polynomials divide monic polynomials.
Or is this only stated for fields? Then we should generalise that lemma to integral domains.

Reply:
The answer to the three comments is the same: they hold for GCD domains, with essentially the same proof. (Do you think divisibility holds over any integral domain? I didn't really thought about that). 

That's why I proved `minimal_polynomial.gcd_domain_eq_field_fractions`, but in practice I am not able to apply it to `ℤ` and `ℚ`. We discussed about that on Zulip, [here](https://leanprover.zulipchat.com/#narrow/stream/116395-maths/topic/Minimal.20polynomial.20over.20.E2.84.9A.20vs.20over.20.E2.84.A4). The problem seems to be in how localization is written, with this `codomain` trick. I tried to improve it but it is going to take forever for me, that's why I decided to PR this anyway. An intermediate solution would be to prove the lemma for a generic polynomial, but still for `ℤ` and `ℚ`.

Reply:
I think that for now I like the intermediate solution. Please add a TODO about the potential generalisation next to those lemmas.

Reply:
Indeed it is not true in general: `x ^ 2 -x - 1` and `x ^ 2 + x -2 -√5` both have `(1 + √5)/2` as root (I am taking `ℤ[√5]` as base ring). We should really say "a minimal polynomial", not "the minimal polynomial"...

Reply:
OK, I will open a separate PR.